crates/holo_hash/src/hash_b64.rs

```diff
@@ -0,0 +1,107 @@
+//! Implements base-64 de/serialization for HoloHash
+// NB: this module has a lot of copy-and-paste from `ser.rs`
+
+use super::*;
+use crate::HoloHash;
+use crate::{error::HoloHashResult, HashType};
+
+/// A wrapper around HoloHash to denote that deserialization should use
+/// base-64 strings rather than raw byte arrays
+//
+// TODO: make HoloHash and HoloHashB64 both deserialize identically (can take either string or seq),
+//       so that the only difference is how they serialize
+#[derive(
+    Debug,
+    Clone,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    derive_more::Constructor,
+    derive_more::Display,
+    derive_more::From,
+    derive_more::Into,
+    derive_more::AsRef,
+)]
+pub struct HoloHashB64<T: HashType>(HoloHash<T>);
```

Comment:
I'm not against this wrapper type... but if something like the `#[serde(with = "serde_bytes")]` function that data could be annotated with would be easier to use in certain places?

Reply:
Totally...I couldn't clearly see how to make that work though, especially when the type involves e.g. `Vec<HeaderHash>` -- you can't annotate the `Vec` so you'd have to create a newtype anyway, I think.